src/Testcontainers.DynamoDB/DynamoDBBuilder.cs

```diff
@@ -0,0 +1,86 @@
+namespace Testcontainers.DynamoDB;
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+[PublicAPI]
+public sealed class DynamoDBBuilder : ContainerBuilder<DynamoDBBuilder, DynamoDBContainer, DynamoDBConfiguration>
+{
+    public const string DynamoDbImage = "amazon/dynamodb-local:1.21.0";
+
+    public const ushort DynamoDbPort = 8000;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DynamoDBBuilder" /> class.
+    /// </summary>
+    public DynamoDBBuilder()
+        : this(new DynamoDBConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DynamoDBBuilder" /> class.
+    /// </summary>
+    /// <param name="dockerResourceConfiguration">The Docker resource configuration.</param>
+    private DynamoDBBuilder(DynamoDBConfiguration dockerResourceConfiguration)
+        : base(dockerResourceConfiguration)
+    {
+        DockerResourceConfiguration = dockerResourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override DynamoDBConfiguration DockerResourceConfiguration { get; }
+    
+
+    /// <inheritdoc />
+    public override DynamoDBContainer Build()
+    {
+        Validate();
+        return new DynamoDBContainer(DockerResourceConfiguration, TestcontainersSettings.Logger);
+    }
+
+    /// <inheritdoc />
+    protected override DynamoDBBuilder Init()
+    {
+        return base.Init()
+            .WithImage(DynamoDbImage)
+            .WithPortBinding(DynamoDbPort, true)
+            .WithWaitStrategy(Wait.ForUnixContainer().AddCustomWaitStrategy(new WaitUntil(DynamoDbPort)));
```

Comment:
```suggestion
            .WithWaitStrategy(Wait.ForUnixContainer().UntilHttpRequestIsSucceeded(request => request.ForPath("/").ForPort(DynamoDbPort))
```

Then you can remove the `WaitUntil` implementation below.